packages/api/src/tasks/providers/default.ts

```diff
@@ -1,15 +1,106 @@
+import { Tile } from '@hyperdx/common-utils/dist/types';
 import mongoose from 'mongoose';
 import ms from 'ms';
 import { URLSearchParams } from 'url';
 
 import * as config from '@/config';
 import { LOCAL_APP_TEAM } from '@/controllers/team';
 import { connectDB, mongooseConnection } from '@/models';
-import Alert from '@/models/alert';
-import { ISavedSearch } from '@/models/savedSearch';
+import Alert, { AlertSource, type IAlert } from '@/models/alert';
+import Connection, { IConnection } from '@/models/connection';
+import Dashboard from '@/models/dashboard';
+import { type ISavedSearch, SavedSearch } from '@/models/savedSearch';
+import { type ISource, Source } from '@/models/source';
+import {
+  type AlertDetails,
+  type AlertProvider,
+  type AlertTask,
+  AlertTaskType,
+} from '@/tasks/providers';
 import { convertMsToGranularityString } from '@/utils/common';
+import logger from '@/utils/logger';
 
-import { AlertProvider, AlertTask } from './index';
+async function getSavedSearchDetails(
+  alert: IAlert,
+): Promise<[IConnection, AlertDetails] | []> {
+  const savedSearchId = alert.savedSearch;
+  const savedSearch = await SavedSearch.findById(savedSearchId).populate<
+    Omit<ISavedSearch, 'source'> & { source: ISource }
+  >('source');
+
+  if (!savedSearch) {
+    logger.error(`savedSearch not found: id=${savedSearchId}`);
+    return [];
+  }
+
+  const { source } = savedSearch;
+  const connId = source.connection;
+  const conn = await Connection.findById(connId);
+  if (!conn) {
+    logger.error(
+      `connection not found: alertId=${alert._id}, connId=${connId}, savedSearchId=${savedSearchId}`,
+    );
+    return [];
+  }
+
+  return [
+    conn,
+    {
+      alert,
+      source,
+      taskType: AlertTaskType.SAVED_SEARCH,
+      savedSearch,
+    },
+  ];
+}
+
+async function getTileDetails(
+  alert: IAlert,
+): Promise<[IConnection, AlertDetails] | []> {
+  const dashboardId = alert.dashboard;
+  const tileId = alert.tileId;
+
+  const dashboard = await Dashboard.findById(dashboardId);
+  if (!dashboard) {
+    logger.error(`dashboard not found: id=${dashboardId}`);
+    return [];
+  }
+
+  const tile = dashboard.tiles?.find((t: Tile) => t.id === tileId);
+  if (!tile) {
+    logger.error(
+      `tile matching alert not found: tile=${tileId},alert=${alert._id}`,
+    );
+    return [];
+  }
+
+  const source = await Source.findById(tile.config.source).populate<
```

Comment:
I don't think we can trust the source property here since I think this is user-submitted, ditto on connections